src/app/loadout/ModPicker.tsx

```diff
@@ -199,6 +200,23 @@ function ModPicker({ plugSets, lockedMods, initialQuery, onAccept, onClose }: Pr
     [lockedMods, plugSets]
   );
 
+  // Now we populate the plugsets with their corresponding plugs.
+  // Due to artificer plugsets being a subset of the corresponding bucket specific plugsets
+  // we sort the plugsets in reverse order by length, so we populate artificer sets first.
+  plugSets.sort((a, b) => b.plugs.length - a.plugs.length);
+  for (const intialPlug of visibleSelectedMods) {
+    const possiblePlugSets = plugSets.filter((set) =>
+      set.plugs.some((plug) => plug.hash === intialPlug.hash)
+    );
+
+    for (const possiblePlugSet of possiblePlugSets) {
+      if (possiblePlugSet.selected.length < possiblePlugSet.maxSelectable) {
+        possiblePlugSet.selected.push(intialPlug);
+        break;
+      }
+    }
+  }
```

Comment:
This ends up modifying the `plugSets` array in the props, which has bitten me before and sounds like something that could cause issues (both the in-place sort and the pushes to `selected`, the pushes are also not idempotent!).

Reply:
That is a good point, the `plugSets` in this instance come from `mapStateToProps` but this should also be living in there. I will pull it up.

Reply:
Done, have another look for me when you get the chance.